decisions/C-ManagerInterface.md

```diff
@@ -0,0 +1,205 @@
+    <method arguments>...)
+```
+
+This has the advantage of reading similarly to a C++ function call,
+with return values first, followed by the `this` pointer (handle), 
+then function arguments.
+
+Alternative, perhaps equally valid, argument orderings are of course
+possible.
+
+## ManagerInterface suite
+
+For our oversimplified `ManagerInterface` whose only methods are 
+`identifier` and `displayName`
+
+```c++
+typedef struct {
+  void (*dtor)(OPENASSETIO_NS(managerAPI_ManagerInterface_h));
+
+  int (*identifier)(OPENASSETIO_NS(SimpleString) *,
+                    OPENASSETIO_NS(SimpleString) *,
+                    OPENASSETIO_NS(managerAPI_ManagerInterface_h));
+
+  int (*displayName)(OPENASSETIO_NS(SimpleString) *,
+                     OPENASSETIO_NS(SimpleString) *,
+                     OPENASSETIO_NS(managerAPI_ManagerInterface_h));
+} OPENASSETIO_NS(managerAPI_ManagerInterface_s);
+
+```
+
+Note that in this case there is a `dtor` (destructor) function but no
+corresponding constructor. For the particular case of `ManagerInterface`
+we would expect a manager plugin to provide a `ManagerInterface_h`
+handle (the details of the plugin system are out of scope here).
+
+## C++ wrapper
+
+A sketch is shown below. For brevity only the `identifier` member 
+function is shown since the implementation of `identifier` and 
+`displayName` are almost identical.
+
+To translate an error coming from C we can define a common mapping of
+error codes/messages to C++ exceptions. Below this is assumed to be
+implemented in a `throwIfError` function (omitted for brevity).
+
+```c++
+constexpr size_t kStringBufferSize = 500;
+
+// Constructor expects to be supplied a valid handle and suite.
+CManagerInterface::CManagerInterface(
+    OPENASSETIO_NS(managerAPI_ManagerInterface_h) handle,
+    OPENASSETIO_NS(managerAPI_ManagerInterface_s) suite)
+    : handle_{handle}, suite_{suite} {}
+
+// Destructor calls suite's `dtor` function.
+CManagerInterface::~CManagerInterface() { suite_.dtor(handle_); }
+
+Str CManagerInterface::identifier() const {
+  // Buffer for error message.
+  char errorMessageBuffer[kStringBufferSize];
+  // Error message.
+  OPENASSETIO_NS(SimpleString)
+  errorMessage{kStringBufferSize, errorMessageBuffer, 0};
+
+  // Return value string buffer.
+  char outBuffer[kStringBufferSize];
+  // Return value.
+  OPENASSETIO_NS(SimpleString) out{kStringBufferSize, outBuffer, 0};
+
+  // Execute corresponding suite function.
+  const int errorCode = suite_.identifier(&errorMessage, &out, handle_);
+
+  // Convert error code/message to exception.
+  throwIfError(errorCode, errorMessage);
+
+  return {out.buffer, out.usedSize};
+}
+```
+
+## Directory structure
+
+The proposed directory structure is the following
+```
+src/openassetio-core-c
+├── CMakeLists.txt
+├── include
+│   └── openassetio
+│       └── c
+│           ├── errors.h
+│           ├── managerAPI
+│           │   └── ManagerInterface.h
+│           ├── ns.h
```

Comment:
maybe call this one namespace.h? ns.h makes me wonder if it is full of ObjC stuff, since every Apple Foundation class is named NSSomething.

Reply:
Renamed as suggested in 4d4b589a